src/material/icon/_icon-theme.scss

```diff
@@ -44,24 +51,33 @@
   }
 }
 
+/// Outputs typography theme styles for the mat-icon.
+/// @param {Map} $theme The theme to generate typography styles for.
 @mixin typography($theme) {
   @if inspection.get-theme-version($theme) == 1 {
     @include _theme-from-tokens(inspection.get-theme-tokens($theme, typography));
   }
   @else {}
 }
 
+/// Outputs density theme styles for the mat-icon.
+/// @param {Map} $theme The theme to generate density styles for.
 @mixin density($theme) {
   @if inspection.get-theme-version($theme) == 1 {
     @include _theme-from-tokens(inspection.get-theme-tokens($theme, density));
   }
   @else {}
 }
 
-@mixin theme($theme) {
+/// Outputs all theme styles for the mat-icon.
+/// @param {Map} $theme The theme to generate styles for.
+/// @param {ArgList} Additional optional arguments (only supported for M3 themes):
+///   $color-variant: The color variant to use for the icon: primary, secondary, tertiary, or error
```

Comment:
I'd rather keep icon for clarity, for some components I think it could be more specific than just "this component", e.g. for select it might say "the selected value highlight"